gcloud/rest/auth/token.py

```diff
@@ -1,95 +1,199 @@
+"""Google Cloud auth via service account file"""
 import datetime
+import enum
 import json
-import logging
 import os
 import threading
 import time
+from typing import Any
+from typing import Dict
+from typing import List  # pylint: disable=unused-import
+from typing import Optional
+
 try:
     from urllib.parse import urlencode
+    from urllib.parse import quote_plus
 except ImportError:
     from urllib import urlencode
+    from urllib import quote_plus
 
-# N.B. the cryptography library is required when calling jwt.encrypt() with
-# algorithm='RS256'. It does not need to be imported here, but this allows us
-# to throw this error at load time rather than lazily during normal operations,
-# where plumbing this error through will require several changes to otherwise-
-# good error handling.
-import cryptography  # pylint: disable=unused-import
```

Comment:
Let's keep this (and the comment explaining why we're keeping it!)

Reply:
Ahh! good point! will do!